include/zephyr/service.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2024 Tomasz Bursztyka.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SERVICE_H_
+#define ZEPHYR_INCLUDE_SERVICE_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include <zephyr/sys/util.h>
+#include <zephyr/toolchain.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup service System service
+ * @ingroup os_services
+ *
+ * A system service is a one-shot initialized software part, instantiated
```

Comment:
Component might sound a bit better here. Just a suggestion.

```suggestion
 * A system service is a one-shot initialized software component, instantiated
```